block/deserialize.go

```diff
@@ -0,0 +1,341 @@
+package block
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/vulpemventures/go-elements/internal/bufferutil"
+
+	"github.com/vulpemventures/go-elements/transaction"
+)
+
+func deserialize(buf *bytes.Buffer) (*Block, error) {
+	header, err := DeserializeHeader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	transactions, err := DeserializeTransactions(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Block{
+		Header: header,
+		TransactionsData: &Transactions{
+			Transactions: transactions,
+		},
+	}, nil
+}
+
+func DeserializeHeader(
+	buf *bytes.Buffer,
+) (*Header, error) {
+	d := bufferutil.NewDeserializer(buf)
+
+	version, err := d.ReadUint32()
+	if err != nil {
+		return nil, err
+	}
+
+	isDyna := false
+	if version>>31 == 1 {
+		isDyna = true
+	}
+	version = blockVersion
+
+	prevBlockHash, err := d.ReadSlice(hashSize)
+	if err != nil {
+		return nil, err
+	}
+
+	merkleRoot, err := d.ReadSlice(hashSize)
+	if err != nil {
+		return nil, err
+	}
+
+	timestamp, err := d.ReadUint32()
+	if err != nil {
+		return nil, err
+	}
+
+	blockHeight, err := d.ReadUint32()
+	if err != nil {
+		return nil, err
+	}
+
+	extData, err := deserializeExtData(d, isDyna)
+	if err != nil {
+		return nil, err
+	}
+
+	extData.IsDyna = isDyna
```

Comment:
couldn't this be done into deserializeExtData? Like:
```
return &ExtData{
	Proof:             proof,
	DynamicFederation: dynamicFederation,
        IsDyna: isDyna,
}
```

Reply:
Done